plutus-emulator/src/Wallet/Emulator/MultiAgent.hs

```diff
@@ -80,22 +81,47 @@ walletEvent w = prism' (WalletEvent w) (\case { WalletEvent w' c | w == w' -> Ju
 chainIndexEvent :: Wallet.Wallet -> Prism' EmulatorEvent ChainIndex.ChainIndexEvent
 chainIndexEvent w = prism' (ChainIndexEvent w) (\case { ChainIndexEvent w' c | w == w' -> Just c; _ -> Nothing })
 
--- | The type of actions in the emulator. @n@ is the type (usually a monad implementing 'WalletAPI') in
--- which wallet actions take place.
+type EmulatedWalletEffects =
+        '[ Wallet.WalletEffect
+         , Error WAPI.WalletAPIError
+         , Wallet.NodeClientEffect
+         , Wallet.ChainIndexEffect
+         , Wallet.SigningProcessEffect
+         ]
+
+type EmulatedWalletControlEffects =
+        '[ Error WAPI.WalletAPIError
```

Comment:
Hm, maybe this should be a different kind of error for control errors?

Reply:
Turns out the `Error` effect isn't needed at all for the control interface, soI just removed it.